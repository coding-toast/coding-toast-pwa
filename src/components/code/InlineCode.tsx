import * as React from 'react';

interface IInlineCodeProps {
  value: string;
}

const InlineCode: React.FC<IInlineCodeProps> = (props) => {
  return (
    <code
      className='px-2 py-1'
      style={{
        color: '#c5c8c6',
        backgroundColor: '#333',
        textShadow: 'text-shadow:0 1px rgba(0, 0, 0, 0.3)',
        background: '#1d1f21',
        borderRadius: '0.3em'
      }}
    >
      {props.value}
    </code>
  );
};

export default InlineCode;
