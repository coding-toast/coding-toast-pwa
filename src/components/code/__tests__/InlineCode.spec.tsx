import * as React from 'react';
import InlineCode from '../InlineCode';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<InlineCode />', () => {
  it('Should render properly', () => {
    const wrapper = shallow(<InlineCode value='<pre><code>console.log("Hello World!")' />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
