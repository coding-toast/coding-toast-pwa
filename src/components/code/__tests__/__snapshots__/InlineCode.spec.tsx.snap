// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineCode /> Should render properly 1`] = `
<code
  className="px-2 py-1"
  style={
    Object {
      "background": "#1d1f21",
      "backgroundColor": "#333",
      "borderRadius": "0.3em",
      "color": "#c5c8c6",
      "textShadow": "text-shadow:0 1px rgba(0, 0, 0, 0.3)",
    }
  }
>
  &lt;pre&gt;&lt;code&gt;console.log("Hello World!")
</code>
`;
